File: uikit/src/main/java/io/agora/chat/uikit/provider/EaseSettingsProvider.java
package io.agora.chat.uikit.provider;


import io.agora.chat.ChatMessage;

/**
 * new message options provider
 *
 */
public interface EaseSettingsProvider {
    /**
     * Whether to allow message reminders
     * @param message
     * @return
     */
    boolean isMsgNotifyAllowed(ChatMessage message);

    /**
     * Whether to set the sound
     * @param message
     * @return
     */
    boolean isMsgSoundAllowed(ChatMessage message);

    /**
     * Whether to allow vibration
     * @param message
     * @return
     */
    boolean isMsgVibrateAllowed(ChatMessage message);

    /**
     * Whether to use the speaker to play sound
     * @return
     */
    boolean isSpeakerOpened();
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/provider/EaseSettingsProvider.kt
package io.agora.uikit.provider

import io.agora.uikit.common.ChatMessage

/**
 * new message options provider
 *
 */
interface EaseSettingsProvider {
    /**
     * Whether to allow message reminders
     * @param message
     * @return
     */
    fun isMsgNotifyAllowed(message: ChatMessage?): Boolean

    /**
     * Whether to set the sound
     * @param message
     * @return
     */
    fun isMsgSoundAllowed(message: ChatMessage?): Boolean

    /**
     * Whether to allow vibration
     * @param message
     * @return
     */
    fun isMsgVibrateAllowed(message: ChatMessage?): Boolean

    /**
     * Whether to use the speaker to play sound
     * @return
     */
    val isSpeakerOpened: Boolean
}